### PR TITLE
[Merged by Bors] - chore(LinearAlgebra/AffineSpace/AffineSubspace): trivial affine morphisms

### PR DESCRIPTION
**Pull request by eric-wieser**
We already have the `LinearEquiv` versions of these.
This also copies the lemmas around the definition

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean [hunk 1]
```diff
@@ -1607,6 +1642,33 @@ end AffineMap
...
+section ofEq
+variable (S₁ S₂ : AffineSubspace k P₁) [Nonempty S₁] [Nonempty S₂]
+
+attribute [local instance] AffineSubspace.toAddTorsor
+
+/-- Affine equivalence between two equal affine subspace.
+
+This is the affine version of `LinearEquiv.ofEq`. -/
+@[simps linear]
+def ofEq (h : S₁ = S₂) : S₁ ≃ᵃ[k] S₂ where
```
Comment (alreadydone) [inline Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean:1654]: Here the existing names are more chaotic, and `ofEq` is a reasonable choice:
[Equiv.Set.ofEq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Equiv/Set.html#Equiv.Set.ofEq), [RingEquiv.subringCongr](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Subring/Basic.html#RingEquiv.subringCongr), [IntermediateField.equivOfEq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/Adjoin.html#IntermediateField.equivOfEq), [Subalgebra.equivOfEq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Subalgebra/Basic.html#Subalgebra.equivOfEq)

File: Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean [hunk 2]
```diff
@@ -1592,6 +1603,30 @@ end AffineSubspace
...
+section ofLe
+variable {S₁ S₂ : AffineSubspace k P₁} [Nonempty S₁] [Nonempty S₂]
+
+attribute [local instance] AffineSubspace.toAddTorsor
+
+/-- Affine map from a smaller to a larger subspace of the same space.
+
+This is the affine version of `Submodule.ofLe`. -/
+@[simps linear]
+def ofLe (h : S₁ ≤ S₂) : S₁ →ᵃ[k] S₂ where
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean:1616]: #8470

[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean]: I've now renamed the one in this PR too.

**Review by alreadydone — APPROVED** [review]
Thanks!

maintainer merge

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**